test/shmemx/shmemx_malloc_with_hints.c

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (c) 2017 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <shmem.h>
+#include <shmemx.h>
+
+#define N 128
+#define MAX(a,b) ((a) > (b)) ? a : b
+#define WRK_SIZE MAX(N/2+1, SHMEM_REDUCE_MIN_WRKDATA_SIZE)
+
+#define NO_TESTS 4
+
+long pSync[SHMEM_ALLTOALL_SYNC_SIZE];
+int pWrk[WRK_SIZE];
+
+
+int *dst, *src;
+
+int sumtoall_with_malloc_hint(long hint, int mype, int npes)
+{
+    int failed = 0;
+    int i;
+
+    if (mype == 0)
+        printf("Sum reduction operation started\n");
+
+    dst = (int *)shmemx_malloc_with_hints(N * sizeof(*dst), hint);
+    src = (int *)shmemx_malloc_with_hints(N * sizeof(*src), hint);
+
+    for (i = 0; i < npes; i++) {
+        src[i] = mype;
+        dst[i] = -1;
+    }
```

Comment:
`dst` and `src` are allocated `N` size, but data is assigned only for `npes` indices. We should assign all `N` values, I think.

Reply:
aah I had it for N values but got replaced in later changes I think. Thanks for bringing this to my attention!